From f28010891665f1a2ebc9038a203e3ec870ce792c Mon Sep 17 00:00:00 2001
From: Sebastian Sturm <s.sturm@arkona-technologies.de>
Date: Thu, 8 Dec 2022 20:43:29 +0100
Subject: [PATCH 7/7] Initialize state->error_buffer_read

random values of state->error_buffer_read might otherwise
cause memory corruption as we write beyond the stderr receive
buffer, or (if state->error_buffer_read > ERROR_BUFFER_SIZE),
the size argument to recv might overflow to very large unsigned
values, causing the recv call to fail (and hence causing spsupr
to reattempt reading from the stderr fd indefinitely)
---
 src/json.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/json.c b/src/json.c
index a9ec4d84707..374ed1da711 100644
--- a/src/json.c
+++ b/src/json.c
@@ -1086,6 +1086,7 @@ DEFUN ("json-rpc-connection", Fjson_rpc_connection, Sjson_rpc_connection, 1, MAN
       /* TODO: mutex_init could fail */
       state->handle = handle;
       state->done = false;
+      state->error_buffer_read = 0;
       SAFE_FREE ();
       return make_user_ptr (json_rpc_state_free, state);
     }
-- 
2.41.0

